test/suites/dev-tanssi/register-with-relay-proof/test_deregister_with_relay_proof.ts

```diff
@@ -0,0 +1,173 @@
+
+                // These are registered in genesis
+                expect(parasRegistered).to.contain(2000);
+                expect(parasRegistered).to.contain(2001);
+
+                // Set storage of pallet_author_noting and pallet_services_payment to test that it gets deleted later
+                const tx1 = polkadotJs.tx.authorNoting.setAuthor(2000, 1, alice.address, 1);
+                const tx2 = polkadotJs.tx.authorNoting.setAuthor(2001, 1, alice.address, 1);
+                await polkadotJs.tx.sudo.sudo(polkadotJs.tx.utility.batchAll([tx1, tx2])).signAndSend(alice);
+
+                // Credits already exist
+                const credits2000 = (await polkadotJs.query.servicesPayment.blockProductionCredits(2000)).toJSON();
+                expect(credits2000).toBeGreaterThan(0);
+                const credits2001 = (await polkadotJs.query.servicesPayment.blockProductionCredits(2001)).toJSON();
+                expect(credits2001).toBeGreaterThan(0);
+            },
+        });
+
+        it({
+            id: "E04",
+            title: "Checking that de-registering paraIds is possible",
+            test: async function () {
+                await context.createBlock();
+
+                const currentSesssion = await polkadotJs.query.session.currentIndex();
+                const sessionDelay = await polkadotJs.consts.registrar.sessionDelay;
+                const expectedScheduledOnboarding =
+                    BigInt(currentSesssion.toString()) + BigInt(sessionDelay.toString());
+
+                const { relayProofBlockNumber, relayStorageProof } = await fetchStorageProofFromValidationData(
+                    polkadotJs
+                );
+                const tx = polkadotJs.tx.registrar.deregisterWithRelayProof(
+                    2001,
+                    relayProofBlockNumber,
+                    relayStorageProof
+                );
+                await tx.signAndSend(alice);
+
+                await context.createBlock();
+
+                const pendingParas = await polkadotJs.query.registrar.pendingParaIds();
+                expect(pendingParas.length).to.be.eq(1);
+                const sessionScheduling = pendingParas[0][0];
+                const parasScheduled = pendingParas[0][1];
+
+                expect(sessionScheduling.toBigInt()).to.be.eq(expectedScheduledOnboarding);
+
+                // These will be the paras in session 2
+                // TODO: fix once we have types
+                expect(parasScheduled.toJSON()).to.deep.equal([2000]);
+
+                // Checking that in session 2 paras are registered
+                await jumpSessions(context, 2);
+
+                // Expect now paraIds to be registered
+                const parasRegistered = await polkadotJs.query.registrar.registeredParaIds();
+                // TODO: fix once we have types
+                expect(parasRegistered.toJSON()).to.deep.equal([2000]);
```

Comment:
Can we check we get the deposit?

Reply:
that we slash the deposit I mean

Reply:
In this test I cannot do it because that parachain is registered in genesis, so it doesn't have a deposit.

Reply:
mmm it's a pity

Reply:
can we register another chain insted?

Reply:
I registered chain 2003 instead